Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -2227,11 +2229,7 @@ private InputActionReference UpdateReferenceForNewAsset(InputActionReference act
             if (newAction == null)
                 return null;
 
-            var reference = InputActionReference.Create(newAction);
-            if (oldActionEnabled)
-                EnableInputAction(reference);
-
-            return reference;
+            return InputActionReference.Create(newAction);
```

Comment:
Do we not still want the functionality where actions that were enabled before assigning a new asset are still enabled afterwards?

Reply:
Actually no. This was the 'real' bug here. The other changes were for another bug that got me mixed up for a while.

`UpdateReferenceForNewAsset` is called only in conjunction with Action property Setters. These setters call `SwapAction` which also does this Disable/Enable dance. So what was happening was double refCount decrements and then double refCount increments. And in the sample project the double decrement would disabled the control early (took me a while to make a Unit test for exactly this scenario as it needs to be at refCount==2 before hitting this code, otherwise the double decrement wouldn't matter much).

Reply:
Ah, nice!